packages/web-components/src/components/dotcom-shell/dotcom-shell-container.ts

```diff
@@ -0,0 +1,256 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { pickBy } from 'lodash-es';
+import { html, property, customElement, LitElement } from 'lit-element';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings';
+import { BasicLink, BasicLinkSet, MastheadLink } from '../../globals/services-store/types/translateAPI';
+import { USER_AUTHENTICATION_STATUS } from '../../globals/services-store/types/profileAPI';
+import { FOOTER_SIZE } from '../footer/footer';
+import '../footer/footer-container';
+import { MastheadProfileItem } from '../masthead/masthead-container';
+import { LocaleModalLocaleList } from '../locale-modal/locale-modal-composite';
+import './dotcom-shell';
+import styles from './dotcom-shell-container.scss';
+
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+/**
+ * Dotcom shell.
+ *
+ * @element dds-dotcom-shell-container
+ */
+@customElement(`${ddsPrefix}-dotcom-shell-container`)
+class DDSLinkWithIcon extends LitElement {
+  private _footerRenderRoot: Element | null = null;
+
+  private _mastheadRenderRoot: Element | null = null;
+
+  private _createFooterRenderRoot() {
+    const footer = this.ownerDocument.createElement(`${ddsPrefix}-footer-container`);
+    this.parentNode?.insertBefore(footer, this.nextSibling);
+    return footer;
+  }
+
+  private _createMastheadRenderRoot() {
+    const masthead = this.ownerDocument.createElement(`${ddsPrefix}-masthead-container`);
+    this.parentNode?.insertBefore(masthead, this);
+    return masthead;
+  }
+
+  /**
+   * `true` to activate the search box.
```

Comment:
@asudoh one thing we can add to the descriptions is if the attribute is for the masthead or for the footer.

Reply:
Good point @jeffchew, updated.